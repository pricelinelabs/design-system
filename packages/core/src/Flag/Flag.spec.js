import React from 'react'
import { render } from 'testing-library'

import { Flag, theme } from '..'

describe('Flag', () => {
  let consoleError
  beforeEach(() => {
    consoleError = console.error
    console.error = jest.fn()
  })
  afterEach(() => (console.error = consoleError))

  test('renders', () => {
    const json = rendererCreateWithTheme(<Flag />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with width prop', () => {
    const json = rendererCreateWithTheme(<Flag width={256} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('width', '256px')
  })

  test('renders with palette color as color prop', () => {
    const text = 'bla'
    const { getByText } = render(
      <Flag width={256} color='promoPrimary.base'>
        {text}
      </Flag>
    )
    expect(getByText(text)).toHaveStyleRule(
      'background-color',
      theme.colors.purple
    )
  })
})
