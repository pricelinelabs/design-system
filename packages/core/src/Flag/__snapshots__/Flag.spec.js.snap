// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag renders 1`] = `
.c0 {
  margin-left: 0;
}

.c4 {
  margin-right: -8px;
  margin-bottom: -8px;
  width: 4px;
  background-color: #0a0;
  color: #fff;
}

.c6 {
  padding-left: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #0a0;
  color: #fff;
}

.c8 {
  margin-left: -8px;
  width: 18px;
  background-color: #0a0;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  width: 8px;
  height: 8px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background-color: inherit;
  background-image: linear-gradient(45deg,transparent 50%,#060 50%);
  position: absolute;
  bottom: 0;
}

.c9 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  background-color: #0a0;
  border-radius: 0 2px 2px 0;
  -webkit-transform: skew(-14deg);
  -ms-transform: skew(-14deg);
  transform: skew(-14deg);
  position: relative;
  z-index: 1;
}

.c7 {
  font-size: 12px;
  border-radius: 0 0 2px 0;
  background-color: #0a0;
  color: #fff;
  z-index: 2;
}

.c3 {
  position: relative;
}

@media screen and (min-width:32em) {
  .c0 {
    margin-left: -8px;
  }
}

@media screen and (min-width:32em) {
  .c6 {
    padding-left: 16px;
    padding-top: 8px;
    padding-bottom: 8px;
  }
}

@media screen and (max-width:31.99em) {
  .c2 {
    display: none;
  }
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
  >
    <div
      className="c4 c5"
      color="secondary"
      width="4px"
    />
  </div>
  <div
    className="c6 c7"
    color="secondary"
  />
  <div
    className="c8 c9"
    color="secondary"
    width="18px"
  />
</div>
`;

exports[`Flag renders with width prop 1`] = `
.c0 {
  margin-left: 0;
  width: 256px;
}

.c4 {
  margin-right: -8px;
  margin-bottom: -8px;
  width: 4px;
  background-color: #0a0;
  color: #fff;
}

.c6 {
  padding-left: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #0a0;
  color: #fff;
}

.c8 {
  margin-left: -8px;
  width: 18px;
  background-color: #0a0;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  width: 8px;
  height: 8px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background-color: inherit;
  background-image: linear-gradient(45deg,transparent 50%,#060 50%);
  position: absolute;
  bottom: 0;
}

.c9 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  background-color: #0a0;
  border-radius: 0 2px 2px 0;
  -webkit-transform: skew(-14deg);
  -ms-transform: skew(-14deg);
  transform: skew(-14deg);
  position: relative;
  z-index: 1;
}

.c7 {
  font-size: 12px;
  border-radius: 0 0 2px 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  background-color: #0a0;
  color: #fff;
  z-index: 2;
}

.c3 {
  position: relative;
}

@media screen and (min-width:32em) {
  .c0 {
    margin-left: -8px;
  }
}

@media screen and (min-width:32em) {
  .c6 {
    padding-left: 16px;
    padding-top: 8px;
    padding-bottom: 8px;
  }
}

@media screen and (max-width:31.99em) {
  .c2 {
    display: none;
  }
}

<div
  className="c0 c1"
  width={256}
>
  <div
    className="c2 c3"
  >
    <div
      className="c4 c5"
      color="secondary"
      width="4px"
    />
  </div>
  <div
    className="c6 c7"
    color="secondary"
  />
  <div
    className="c8 c9"
    color="secondary"
    width="18px"
  />
</div>
`;
